x/merkledb/db.go

```diff
@@ -1235,40 +1184,15 @@ func (db *merkleDB) getEditableNode(key path) (*node, error) {
 // Editing the returned node affects the database state.
 // Returns database.ErrNotFound if the node doesn't exist.
 // Assumes [db.lock] is read locked.
-func (db *merkleDB) getNode(key path) (*node, error) {
+func (db *merkleDB) getNode(key path, hasValue bool) (*node, error) {
 	if db.closed {
 		return nil, database.ErrClosed
 	}
 	if key == RootPath {
 		return db.root, nil
 	}
-
-	if n, isCached := db.nodeCache.Get(key); isCached {
-		db.metrics.DBNodeCacheHit()
-		if n == nil {
-			return nil, database.ErrNotFound
-		}
-		return n, nil
-	}
-
-	db.metrics.DBNodeCacheMiss()
-	db.metrics.IOKeyRead()
-	rawBytes, err := db.nodeDB.Get(key.Bytes())
-	if err != nil {
-		if err == database.ErrNotFound {
-			// Cache the miss.
-			if err := db.nodeCache.Put(key, nil); err != nil {
-				return nil, err
-			}
-		}
-		return nil, err
-	}
-
-	node, err := parseNode(key, rawBytes)
-	if err != nil {
-		return nil, err
+	if hasValue {
+		return db.valueNodesDB.Get(key)
```

Comment:
Can we make these in to a  `switch`? In general if there are more than 2 if statements that return in a row I use one